e3sm_diags/e3sm_diags_driver.py

```diff
@@ -40,132 +68,145 @@ def get_default_diags_path(set_name, run_type, print_path=True):
     return pth
 
 
-def _save_env_yml(results_dir):
+def save_provenance(results_dir: str, parser: CoreParser) -> ProvPaths:
+    """
+    Store the provenance in results_dir.
+    """
+    prov_dir = os.path.join(results_dir, "prov")
+
+    paths: ProvPaths = {
+        "log_path": os.path.join(prov_dir, LOG_FILENAME),
+        "parameter_files_path": None,
+        "python_script_path": None,
+        "env_yml_path": None,
+        "index_html_path": None,
+    }
+
+    paths["parameter_files_path"] = _save_parameter_files(prov_dir, parser)
+    paths["python_script_path"] = _save_python_script(prov_dir, parser)
```

Comment:
Prov paths are now captured here to consolidate them under `_log_diagnostic_run_info()`.